components/annotationList.js

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import Router from "next/router";
+import { render } from "react-dom";
+
+/**
+ * annotations(annArray) : function that returns 1 array of all the annotation tags
+ *
+ * @param {*} annArray : array of annotations (which each annotation contains an array of tags)
+ */
+function annotations(annArray) {
+  if (Array.isArray(annArray)) {
+    var returnArray = [];
+    return annArray.map(x => returnArray.concat(x.Tags)).flat();
+  } else {
+    console.log(annArray);
+  }
+}
+
+function mapAnnotationsHelper(uniqueArray) {
+  let rows = [];
+  for (var j = 0; j < uniqueArray.length; j++) {
+    rows.push(
+      <div>
+        <a href={"/search/" + uniqueArray[j]}>{uniqueArray[j]}</a>
+      </div>
+    );
+  }
+  return rows;
+}
+
+/**
+ * mapAnnotations(finalArray) : function that gets unique set of all tags and returns them as XML links
+ *
+ * need to fix: doesn't work properly with different cases of tags (e.g. uppercase/lowercase)
+ *
+ * @param {*} finalArray : array of all tags in the array
+ */
+function mapAnnotations(finalArray) {
```

Comment:
I think this deal with if the some tags are redundant and exceed the limited space, right? I would not care about this now as I see it as edge case. But good that you have started thinking about it 👍 . Let us for now commit out this function and just display non redundant annotations by using Set, just like your code does and then return it.